jakarta-eclipse/src/org/jakarta/lsp4e/JakartaLanguageClient.java

```diff
@@ -78,4 +90,30 @@ public boolean isCanceled() {
  			return JDTServicesManager.getInstance().getExistingContextsFromClassPath(uri, snippetContexts);
  		});
  	}
+ 	
+	@Override
+	public CompletableFuture<List<Either<Command, CodeAction>>> getCodeAction(CodeActionParams params) {
+		return codeAction(params)
+				.thenApply(codeActions ->{
+					return codeActions.stream()
+							.map(ca -> {
+								Either<Command, CodeAction> e = Either.forRight(ca);
+								return e;
+							}.collect(Collectors.toList()));
+				});
+	}
+
+	public CompletableFuture<List<CodeAction>> codeAction(CodeActionParams params){
+		IProgressMonitor monitor = new NullProgressMonitor();
+		JDTUtils utils = new JDTUtils();
+		JakartaJavaCodeActionParams JakartaParams = new JakartaJavaCodeActionParams(params.getTextDocument(), params.getRange(), params.getContext());
+		return CompletableFutures.computeAsync((cancelChecker) -> {
+			try {
+				return JDTServicesManager.getInstance().getCodeAction(JakartaParams, utils, monitor);
+			} catch (JavaModelException e) {
+				return null;
+			}
+		});
+	}
+	
```

Comment:
I have codeAction return CompletableFuture<List<CodeAction>> , then map it into CompletableFuture<List<Either<Command, CodeAction>>> in getCodeAction

Problem is, the ca has error, saying it has to be a functional interface. I don't see the sample code has set ca as an interface instead it is a method inside a class.

Any advices? @kathrynkodama @rzgry